### PR TITLE
3dParametricCurveDigitizer

### PR DESCRIPTION
**Pull request by copyme**
A tool for digitization of 3D parametric curves (needs #1339).

### CODE REVIEW

**Comment by copyme** [comment]
For an example see a screenshot in my comment to #339

**Comment by copyme** [comment]
image


Rotated and digitized knot 5_2 that was visualized with 3dCurveTangentEstimator.

**Comment by dcoeurjo** [comment]
can you please merge back the master ?

**Comment by copyme** [comment]
@dcoeurjo done

**Comment by kerautret** [comment]
thanks also @copyme conflict done

File: generators/3dParametricCurveDigitizer.cpp [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+	Knot_6_2 Parametric knot 6_2 (polynomial knot)
+		Required parameter(s): --param1, --param2, --param3
+	Knot_7_4 Parametric knot 7_4 (polynomial knot)
+		Required parameter(s): --param1, --param2, --param3
+ @endcode
+
+ @b Example:
+ @code
+ # generate a trefoli knot:
+ 3dParametricCurveDigitizer --curve Knot_3_1 --param1 10 --param2 10 --param3 10 --tstart -2.2 --tend 2.2 --step 0.001 --output knot_3_1
```
Comment (kerautret) [inline generators/3dParametricCurveDigitizer.cpp]: I added the example also in the command line help to have a direct way to check the tool.

**Comment by kerautret** [comment]
👍 wainting for travis and merging

**Comment by copyme** [comment]
@kerautret thanks!

File: generators/3dParametricCurveDigitizer.cpp (hunk 1 above)
Comment (copyme) [reply generators/3dParametricCurveDigitizer.cpp]: cool, thanks

**Pull request merged**